nifi-nar-bundles/nifi-azure-bundle/nifi-azure-services-api/pom.xml

```diff
@@ -36,6 +36,16 @@
             <groupId>org.apache.nifi</groupId>
             <artifactId>nifi-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.microsoft.azure.kusto</groupId>
+            <artifactId>kusto-data</artifactId>
+            <version>5.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>com.microsoft.azure.kusto</groupId>
+            <artifactId>kusto-ingest</artifactId>
+            <version>5.0.1</version>
+        </dependency>
```

Comment:
Are these additions necessary at the API package level? It would be better to structure the interface to avoid the need for these dependencies if possible.

Reply:
yeah, I struggled to decouple the dependencies of azure-kusto-java sdk from API package level. The interfaces KustoIngestService and KustoQueryService are decoupled from kusto java SDK dependency. 
But had to include dependencies in the API package level since we had com.microsoft.azure.kusto.ingest.IngestionProperties while submitting an ingestion request to Kusto cluster and com.microsoft.azure.kusto.data.KustoResultSetTable as a part ingestionResult from kusto after ingestion.